src/rust/ensogl/lib/components/src/list_view/entry.rs

```diff
@@ -20,266 +19,175 @@ pub const PADDING:f32 = 14.0;
 }
 
-impl SingleMaskedProvider {
-
-    /// Return the index to the unmasked underlying data. Will only be valid to use after
-    /// calling `clear_mask`.
-    ///
-    /// Transform index of an element visible in the menu, to the index of the all the objects,
-    /// accounting for the removal of the selected item.
-    ///
-    /// Example:
-    /// ```text
-    /// Mask              `Some(1)`
-    /// Masked indices    [0,     1, 2]
-    /// Unmasked Index    [0, 1,  2, 3]
-    /// -------------------------------
-    /// Mask              `None`
-    /// Masked indices    [0, 1, 2, 3]
-    /// Unmasked Index    [0, 1, 2, 3]
-    /// ```
-    pub fn unmasked_index(&self, ix:Id) -> Id {
-        match self.mask.get() {
-            None                 => ix,
-            Some(id) if ix < id  => ix,
-            Some(_)              => ix+1,
-        }
-    }
 
-    /// Mask out the given index. All methods will now skip this item and the `SingleMaskedProvider`
-    /// will behave as if it was not there.
-    ///
-    /// *Important:* The index is interpreted according to the _masked_ position of elements.
-    pub fn set_mask(&self, ix:Id) {
-        let internal_ix = self.unmasked_index(ix);
-        self.mask.set(Some(internal_ix));
-    }
+// === Entry Provider for Vectors of Strings ===
 
-    /// Mask out the given index. All methods will now skip this item and the `SingleMaskedProvider`
-    /// will behave as if it was not there.
-    ///
-    /// *Important:* The index is interpreted according to the _unmasked_ position of elements.
-    pub fn set_mask_raw(&self, ix:Id) {
-        self.mask.set(Some(ix));
-    }
+#[derive(Debug)]
+struct StringEntry {
+    display_object : display::object::Instance,
+    label          : text::Area,
+}
+
+impl StringEntry {
+    fn new(app:&Application, string:&str) -> Self {
+        let logger = Logger::new("StringEntry");
+        let display_object = display::object::Instance::new(logger);
 
-    /// Clear the masked item.
-    pub fn clear_mask(&self) {
-        self.mask.set(None)
+        let label = text::Area::new(app);
+        label.add_to_scene_layer(&app.display.scene().layers.label);
+        display_object.add_child(&label);
+        let styles = StyleWatch::new(&app.display.scene().style_sheet);
+        let text_color = styles.get_color(ensogl_theme::widget::list_view::text);
+        label.set_default_color(text_color);
+        label.set_default_text_size(text::Size(LABEL_SIZE));
+        label.set_position_xy(Vector2(PADDING,LABEL_SIZE/2.0));
+        label.set_content(string);
+
+        Self {display_object,label}
     }
 }
 
-impl From<AnyModelProvider> for SingleMaskedProvider {
-    fn from(content:AnyModelProvider) -> Self {
-        let mask = default();
-        SingleMaskedProvider{content,mask}
+impl display::Object for StringEntry {
+    fn display_object(&self) -> &display::object::Instance {
+        &self.display_object
     }
 }
 
+impl Entry for StringEntry {
```

Comment:
there are no secitons in the file for these entires / providers

Reply:
There are sections now.